macrosAutoCatsTests/src/test/resources/test-cases/chainingFactoryMethods.success

```diff
@@ -22,4 +22,4 @@ require(theC.b != null)
 
 require(created.size == 2)
 require(created.contains("a"))
-require(created.contains("B"))
+require(created.contains("b"))
```

Comment:
Probably not very relevant, but to test the ordering that we care about - that is, the ordering of flatMaps, not of method calls - in the factory methods that return resources, I think it would be better to populate the `created` collection as part of resource allocation/cleanup, e.g. `Resource.make(IO { created.add("b"); B() })(_ => IO.unit)`